src/morphodict/lexicon/management/commands/importjsondict.py

```diff
@@ -0,0 +1,174 @@
+
+            if existing := Wordform.objects.filter(slug=entry["slug"]).first():
+                # Cascade should take care of all related objects.
+                existing.delete()
+
+            wf = Wordform.objects.create(
+                text=entry["head"],
+                raw_analysis=entry.get("analysis", None),
+                paradigm=entry.get("paradigm", None),
+                slug=entry["slug"],
+                is_lemma=True,
+                linguist_info=entry["linguistInfo"],
+            )
+            wf.lemma = wf
+            wf.save()
+
+            # Instantiate wordforms
+            if wf.analysis and wf.paradigm:
+                for (
+                    prefix_tags,
+                    suffix_tags,
+                ) in paradigm_manager.all_analysis_template_tags(wf.paradigm):
+                    analysis = RichAnalysis((prefix_tags, wf.text, suffix_tags))
+                    for generated in strict_generator().lookup(analysis.smushed()):
+                        # Skip re-instantiating lemma
+                        if analysis == wf.analysis:
+                            continue
+
+                        Wordform.objects.create(
+                            # For now, leaving paradigm and linguist_info empty;
+                            # code can get that info from the lemma instead.
+                            text=generated,
+                            raw_analysis=analysis.tuple,
+                            lemma=wf,
+                            is_lemma=False,
+                        )
+
+            if wf.raw_analysis is None:
+                self.index_unanalyzed_form(wf)
+
+            self.create_definitions(wf, entry["senses"])
+
+            seen_slugs.add(wf.slug)
+
+        for entry in form_definitions:
+            lemma = Wordform.objects.get(slug=entry["formOf"])
+
+            wf, created = Wordform.objects.get_or_create(
+                lemma=lemma, text=entry["head"], raw_analysis=entry["analysis"]
+            )
+
+            self.create_definitions(wf, entry["senses"])
+
+        if purge:
+            rows, breakdown = Wordform.objects.filter(
+                slug__in=existing_slugs - seen_slugs
+            ).delete()
+            if rows:
+                logger.warning(
+                    f"Purged {rows:,} rows from database for existing entries not found in import file: %r",
+                    breakdown,
+                )
+        call_command("builddefinitionvectors")
+        call_command("translatewordforms")
+
+    def create_definitions(self, wordform, senses):
+        keywords = set()
+
+        for sense in senses:
+            d = Definition.objects.create(
+                wordform=wordform,
+                text=sense["definition"],
+            )
+            for source in sense["sources"]:
+                try:
+                    d.citations.add(source)
+                except:
+                    breakpoint()
+                    raise
```

Comment:
Leftover debugging code? What happened here?

```suggestion
                d.citations.add(source)
```

Reply:
Yup, exactly.